File: sp48/crypt/prohorow/test5.c
/****************************************************************************/
/*                                                                          */
/*      Data Encryption ToolKit. Version 1.3                                */
/*                                                                          */
/*      test5.c -- test module for DES module 5                             */
/*                                                                          */
/*      Copyright (c) 1991, 1992, Andrew Prokhorow. All rights reserved.    */
/*                                                                          */
/*      Purpose:                                                            */
/*        Test password_decrypt_program function. File test5.exe must be    */
/*        encrypted from test to main and from test_value to incorrect      */
/*        by arbitrary key (password).                                      */
/*                                                                          */
/****************************************************************************/


#include <stdio.h>
#include "encrypt.h"

long test_value = 0;
char correct[] = "Password correct\n";
char incorrect[] = "Password incorrect\n";

void test () {
  printf (correct);
  }

int main () {
  password_decrypt_program ();
  if (test_value) {
    printf (incorrect);
    return 1;
    }
  test ();
  return 0;
  }

File: sp48/crypt/prohorow/test5.pas
{****************************************************************************}
{*                                                                          *}
{*      Data Encryption ToolKit. Version 1.3                                *}
{*                                                                          *}
{*      test5.pas -- test module for DES module 5                           *}
{*                                                                          *}
{*      Copyright (c) 1991, 1992, Andrew Prokhorow. All rights reserved.    *}
{*                                                                          *}
{*      Purpose:                                                            *}
{*        Test PasswordDecryptProgram procedure. File test5.exe must be     *}
{*        encrypted from Test to @ and from TestValue to Incorrect          *}
{*        by arbitrary key (password).                                      *}
{*                                                                          *}
{****************************************************************************}


program Test5;

  uses Encrypt;

  const
    TestValue: LongInt = 0;
    Correct: String = 'Password correct';
    Incorrect: String = 'Password incorrect';

  procedure Test;
    begin
      WriteLn (Correct)
    end;

  begin
    PasswordDecryptProgram;
    if TestValue <> 0 then
      begin
        WriteLn (Incorrect);
        Halt (1)
      end;
    Test;
    Halt (0)
  end.
